include/rdma/fabric.h

```diff
@@ -402,6 +402,49 @@ struct fi_info {
 	struct fi_fabric_attr	*fabric_attr;
 };
 
+struct fi_device_attr {
+	char			*name;
+	char			*device_id;
+	char			*device_version;
+	char			*vendor_id;
```

Comment:
Having strings only for vendor_id/device_id might be annoying if somebody ever wants to match a specific vendor/device since vendor names could change (when companies get bought/renamed/etc). Unfortunately, vendor/device numeric IDs seem PCI-specific (at least they are different for USB).

Reply:
I personally would rather see "Intel" or "Cisco" than 8086 or ...  If the vendor name can change, changing the id seems just as likely.

Reply:
If it's only for printing messages, it doesn't matter much. If anybody ever wants to do specific for a given vendor, it's harder.
Numeric IDs cannot change without breaking kernel drivers. Vendor numeric IDs are assigned by the PCI SIG when vendors register.
However strings reported by lspci are unofficial. They are maintained by the pciutils guys. The "EMC" string changed to "Dell EMC" recently, but "Dell" wasn't renamed to "Dell EMC". When getting strings from something else than pciutils, strings can be different. For instance, it looks like Windows reports "Intel (R)" for 0x8086 while pciutils says "Intel Corporation".